More debugging Meta Powers

nandastone opened the pull request:
This MR improves the Meta Powers UI by:

Adding the option to persist toggle Meta Powers. This helps with development, so when you make a code change and refresh the app you don't need to reselect your Meta Powers.

![CleanShot 2021-12-10 at 16 09 10](https://user-images.githubusercontent.com/199204/145525900-568d2f31-3c9b-449c-bf6f-4ed8c25e3494.png)

Swap the "Reset Cooldowns" to a "Disable Cooldowns" toggle. I found this much easier in testing than needing to re-open the Meta Powers UI and click the "Reset" button every round.

![CleanShot 2021-12-10 at 16 12 02](https://user-images.githubusercontent.com/199204/145526388-a9581e09-49de-4108-833a-15b9adf2084c.gif)

Add a new "Disable Materialization Sickness" toggle. I found this useful in testing when I wanted to summon a creature and test some ability straight away, instead of needing to skip until its next turn.

![CleanShot 2021-12-10 at 16 19 12](https://user-images.githubusercontent.com/199204/145527032-00a3cded-1052-4d62-a81f-8f1310609fd6.gif)

The queue hover state doesn't show where the creature will actually have its first turn, but I don't think that's a huge deal for a testing tool.


<a href="https://gitpod.io/#https://github.com/FreezingMoon/AncientBeast/pull/1943"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>

DreadKnight commented:
> This MR improves the Meta Powers UI by:
> 
> Adding the option to persist toggle Meta Powers. This helps with development, so when you make a code change and refresh the app you don't need to reselect your Meta Powers.
> 

Simply add persistence by default and make "Toggled Powers" text itself act like an unselect everything (a reset basically) ;-)
That way we get some extra very useful utility and with less settings / visual bloat, also while being more consistent overall.

> ![CleanShot 2021-12-10 at 16 09 10](https://user-images.githubusercontent.com/199204/145525900-568d2f31-3c9b-449c-bf6f-4ed8c25e3494.png)
> 
> Swap the "Reset Cooldowns" to a "Disable Cooldowns" toggle. I found this much easier in testing than needing to re-open the Meta Powers UI and click the "Reset" button every round.
> 

Yeah, this makes more sense overall :)

> ![CleanShot 2021-12-10 at 16 12 02](https://user-images.githubusercontent.com/199204/145526388-a9581e09-49de-4108-833a-15b9adf2084c.gif)
> 
> Add a new "Disable Materialization Sickness" toggle. I found this useful in testing when I wanted to summon a creature and test some ability straight away, instead of needing to skip until its next turn.
> 

This is definitely something that I bumped into a LOT as well. I was going to open a new issue soon and propose a power, so that when you would select it and click on the avatar of a queued unit, it would trigger a chain reaction to skip to it until active.
Though that type of functionality would be more of a tool overall than something passive or active I guess, but can be useful.

> ![CleanShot 2021-12-10 at 16 19 12](https://user-images.githubusercontent.com/199204/145527032-00a3cded-1052-4d62-a81f-8f1310609fd6.gif)
> 
> The queue hover state doesn't show where the creature will actually have its first turn, but I don't think that's a huge deal for a testing tool.
> 

Yeah, not that of a huge deal right now I guess, though if that improves with time, it would be neat as well. Feel free to open issue.

nandastone commented:
I've updated the UI to always persist, with a "reset" button:

![CleanShot 2021-12-11 at 23 01 09](https://user-images.githubusercontent.com/199204/145677441-b52baca2-a399-45f8-92b3-2edc7e7657ee.gif)

I've updated the queue preview when "Disable Materialization Sickness" is enabled:

![CleanShot 2021-12-11 at 22 55 45@2x](https://user-images.githubusercontent.com/199204/145677343-fd3f40dd-2cfb-46c8-913b-618e12515b1a.png)

> Should be "Is this unit a Dark Priest?" :)

What's the difference between a unit and a creature?

DreadKnight commented:
> I've updated the UI to always persist, with a "reset" button:
> 
> ![CleanShot 2021-12-11 at 23 01 09](https://user-images.githubusercontent.com/199204/145677441-b52baca2-a399-45f8-92b3-2edc7e7657ee.gif)
> 

Looking good!
> I've updated the queue preview when "Disable Materialization Sickness" is enabled:
> 
> ![CleanShot 2021-12-11 at 22 55 45@2x](https://user-images.githubusercontent.com/199204/145677343-fd3f40dd-2cfb-46c8-913b-618e12515b1a.png)
> 

Yey!

> > Should be "Is this unit a Dark Priest?" :)
> 
> What's the difference between a unit and a creature?

Not much in-game, though usually referring to "units" to the ones that are very human like rather then creature. But capitalization was missing there as well xD

DreadKnight commented:
Did more testing, clicking the close upper right close button doesn't work for this, as it throws an error in the console.
Though this might work just fine locally I guess. Esc key works fine and so do the close buttons from the other screens.

Also, the tooltip background in this scenario should probably have less transparency in order to have the text way easier to read overall, as tooltips overlap icons most of the time which have a lot of white, which creates a conflict, causing some bad contrast.

DreadKnight commented:
Can't figure out why I can't easily check out your branches locally in VSCode by simply selecting it from within the UI or searchbox.

nandastone commented:
> Did more testing, clicking the close upper right close button doesn't work for this, as it throws an error in the console. Though this might work just fine locally I guess. Esc key works fine and so do the close buttons from the other screens.
> 
> Also, the tooltip background in this scenario should probably have less transparency in order to have the text way easier to read overall, as tooltips overlap icons most of the time which have a lot of white, which creates a conflict, causing some bad contrast.

Good pickups, fixed the close icon and increased opacity:

![CleanShot 2021-12-13 at 06 17 47](https://user-images.githubusercontent.com/199204/145728094-1b9ef9ff-5236-483b-88f3-93ea1618599c.png)

DreadKnight commented:
> > Did more testing, clicking the close upper right close button doesn't work for this, as it throws an error in the console. Though this might work just fine locally I guess. Esc key works fine and so do the close buttons from the other screens.
> > Also, the tooltip background in this scenario should probably have less transparency in order to have the text way easier to read overall, as tooltips overlap icons most of the time which have a lot of white, which creates a conflict, causing some bad contrast.
> 
> Good pickups, fixed the close icon and increased opacity:
> 
> ![CleanShot 2021-12-13 at 06 17 47](https://user-images.githubusercontent.com/199204/145728094-1b9ef9ff-5236-483b-88f3-93ea1618599c.png)

Tested, looks way better now and close button actually works, woot! 👍🏻 Time to merge this!
